backend/core/src/auth/services/authz.service.ts

```diff
@@ -45,21 +46,46 @@ export class AuthzService {
       //  A User becomes a leasing agent for a given listing if he has a relation (M:N) with it.
       //  User side this is expressed by 'leasingAgentInListings' property.
       await Promise.all(
-        user?.leasingAgentInListings.map((listing: Listing) => {
+        user?.leasingAgentInListings.map((leasingAgentInListing: Listing) => {
           void e.addPermissionForUser(
             user.id,
             "application",
-            `!r.obj || r.obj.listing_id == '${listing.id}'`,
+            `r.obj.listingId == '${leasingAgentInListing.id}'`,
             `(${authzActions.read}|${authzActions.create}|${authzActions.update}|${authzActions.delete})`
           )
           void e.addPermissionForUser(
             user.id,
             "listing",
-            `!r.obj || r.obj.id == '${listing.id}'`,
+            `r.obj.id == '${leasingAgentInListing.id}'`,
             `(${authzActions.read}|${authzActions.update})`
           )
         })
       )
+
+      if (user.roles?.isJurisdictionalAdmin) {
+        await Promise.all(
+          user.jurisdictions.map((adminInJurisdiction: Jurisdiction) => {
+            void e.addPermissionForUser(
+              user.id,
+              "application",
+              `r.obj.jurisdictionId == '${adminInJurisdiction.id}'`,
+              `(${authzActions.read}|${authzActions.create}|${authzActions.update}|${authzActions.delete})`
+            )
+            void e.addPermissionForUser(
+              user.id,
+              "listing",
+              `r.obj.jurisdictionId == '${adminInJurisdiction.id}'`,
+              `(${authzActions.read}|${authzActions.create}|${authzActions.update}|${authzActions.delete})`
+            )
+            void e.addPermissionForUser(
+              user.id,
+              "user",
+              `r.obj.jurisdictionId == '${adminInJurisdiction.id}'`,
+              `(${authzActions.read}|${authzActions.invitePartner}|${authzActions.inviteJurisdictionalAdmin})`
+            )
+          })
+        )
+      }
```

Comment:
The policy itself is good, I mean in this file we're calling `addPermissionForUser` on all of the applicable resource types, when we could just add it for the resource type that is getting requested. So it could be more dynamic like:
```
void e.addPermissionForUser(
  user.id,
  type,
  `r.obj.jurisdictionId == '${adminInJurisdiction.id}'`,
  available permissions passed in instead of hardcoded here
)
```
Same kind of thinking can go with `leasingAgentInListings` and we can update those to only get added if the user has the isPartner role to be more consistent with the `isJurisdictionalAdmin` check.